sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -152,6 +160,31 @@ public Mono<ShouldRetryResult> shouldRetry(Exception e) {
         return this.throttlingRetry.shouldRetry(e);
     }
 
+      private boolean gatewayRequestCanFailoverOnTimeout(RxDocumentServiceRequest request) {
+        //Query Plan requests
+        if(request.getResourceType() == ResourceType.Document
+            && request.getOperationType() == OperationType.QueryPlan) {
+            return true;
+        }
+
+        //Meta data request check
+        boolean isMetaDataRequest = (request.getOperationType() != OperationType.ExecuteJavaScript
+            && request.getResourceType() == ResourceType.StoredProcedure)
+            || request.getResourceType() != ResourceType.Document;
+
+        //Meta Data Read
+        if(isMetaDataRequest && request.isReadOnly()) {
+              return true;
+        }
+
+        //Data Plane Read & Write
+        if(!isMetaDataRequest && !request.isAddressRefresh()) {
```

Comment:
write operation also retriable? (how we know the request is not reached to server?)

Reply:
Yes, data plane writes (meta data writes are not) operations are retriable and should be handled in this case. I believe that because we are getting an error code the request never reached the server.